docs/Contributing/API-for-contributors.md

```diff
@@ -1712,4 +1714,63 @@ Returns the URL to open when clicking the "Transparency" menu item in Fleet Desk
 
 Redirects to the transparency URL.
 
+### Check if an installer exists 
+
+Checks if a pre-built fleet-osquery installer with the given parameters exists.
+
+`HEAD /api/_version_/fleet/download_installer/{enroll_secret}/{kind}`
+
+#### Parameters
+
+| Name          | Type    | In    | Description                                                        |
+| ------------- | ------- | ----- | ------------------------------------------------------------------ |
+| enroll_secret | string  | path  | The global enroll secret.                                          |
+| kind          | string  | path  | The installer kind: pkg, msi, deb or rpm.                          |
+| desktop       | boolean | query | Set to `true` to ask for an installer that includes Fleet Desktop. |
+
+##### Default response
+
+`Status: 200`
+
+If an installer with the provided parameters is found.
+
+##### Installer doesn't exist
+
+`Status: 400`
+
+If an installer with the provided parameters doesn't exist.
+
+### Download an installer 
+
+Downloads a pre-built fleet-osquery installer with the given parameters.
+
+`GET /api/_version_/fleet/download_installer/{enroll_secret}/{kind}`
+
+#### Parameters
+
+| Name          | Type    | In    | Description                                                        |
+| ------------- | ------- | ----- | ------------------------------------------------------------------ |
+| enroll_secret | string  | path  | The global enroll secret.                                          |
+| kind          | string  | path  | The installer kind: pkg, msi, deb or rpm.                          |
+| desktop       | boolean | query | Set to `true` to ask for an installer that includes Fleet Desktop. |
+
+##### Default response
+
+```
+Status: 200
+Content-Type: application/octet-stream
+Content-Disposition: attachment
+Content-Length: <length>
+Body: <blob>
```

Comment:
cc: @lukeheath the new endpoints for the UI, since this is behind auth, you'll have to do something similar to what you do for CSVs:

https://github.com/fleetdm/fleet/blob/44f902218c0c1d7d569e8a7c6884f4f8ea029c6e/frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx#L1326-L1328

but with `type: 'application/octet-stream'` (the content type comes from the response if you want to take it from there)

happy to tweak things, just lmk!

Reply:
Got it, thanks for the heads up!